MigrationScript/Migration.ps1

```diff
@@ -144,7 +181,194 @@ function Main
+	}
+
+	return @{
+		provisiongState = $provisioningState
+	}
+}
+
+<#
+.SYNOPSIS
+Function to migrate the Hana Providers to ams v2
+
+.PARAMETER secretName
+Secret Name, this contains the name of the provider. 
+
+.PARAMETER secretValue
+Secret Value, this contains the provider configuration.
+
+.PARAMETER hostfile
+hostfile object.
+
+.PARAMETER logger
+logger object.
+
+.EXAMPLE
+MigrateNetWeaverProvider -secretName $secret.name -secretValue $secret -logger $logger
+#>
+function MigrateNetWeaverProvider([string]$secretName, $secretValue, $hostfile, $logger) {
+	$parsedArmId = Get-ParsedArmId $amsv2ArmId
+
+	[string]$monitorName = $parsedArmId.amsResourceName;
+	[string]$resourceGroupName = $parsedArmId.amsResourceGroup;
+	[string]$subscriptionId = $parsedArmId.subscriptionId;
+	# set context.
+	Set-AzContext -SubscriptionId $subscriptionId -TenantId $tenantId;
+
+	[string]$providerType = $($secretValue.type);
+	[string]$providerName = $($secretValue.name) # + (Get-Random -Minimum 2 -Maximum 20000).ToString();
+	Write-Host "Provider Name is $($secretValue.name)";
+	$providerProperties = $($secretValue.properties)
+    $metadata = $($secretValue.metadata)
+	$requestObj = @{
+		Name = $providerName
+		body = @{
+			properties = @{
+				providerSettings = @{
+					providerType = "SapNetWeaver"
+					sapHostname = $($providerProperties.sapHostName)
+					sapSid = $($metadata.sapSid)
+					sapInstanceNr = $($providerProperties.sapInstanceNr).ToString()
+					sapHostFileEntries = $hostfile
+				}
+			}
+		}
+	}
+
+	# check provider status before making a put call
+	$getResponse = GetAmsV2ProviderStatus -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerName $providerName -logger $logger;
+	$provisioningState = $getResponse.provisiongState;
+	$logger.LogInfo("Current Provisioning State : $provisioningState")
+
+	if($provisioningState -eq "Succeeded") {
+		$logger.LogInfo("Provider $providerName already exists..");
+		Continue;
+	}
+	elseif ($provisioningState -eq "Failed") {
+		$managedKvName = GetAmsV2ManagedKv -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger;
+		$funcName = GetAmsV2ManagedFunc -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerType $providerType -logger $logger;
+		# TODO : code to generate the key name 
+		# DeleteAndPurgeSecretFromKeyVault -keyVaultName "dummyName" -secretKey "dummyKey"
+	}
+
+	# call the put provider method.
+	PutAmsV2Provider -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -request $requestObj -logger $logger;
+		
+	# we will check the provisioning status for the provider 15 times in 20 sec intervals.
+	$checks = 0;
+
+	# default providioning state is accepted, we will keep checking till is changes.
+	$provisioningState = "Accepted";
+		
+	while ($checks -le 15 -and $provisioningState -eq "Accepted") {
```

Comment:
why 15 ?

Reply:
@M2skills can you take this question?

Reply:
threshold for provider creation is 6 minutes, so 15 checks at 20 sec intervals was used